blaze/expr/collections.py

```diff
@@ -31,19 +33,200 @@
            'isin',
            'Join',
            'join',
+           'Key',
            'Merge',
            'merge',
            'Sample',
            'sample',
            'Shift',
            'shift',
            'Sort',
+           'Sort2',
            'sort',
+           'sort2',
            'Tail',
            'tail',
            'transform']
 
 
+class Key(Expr):
+
+    """Expression sorted in ascending or descending order"""
+
+    _arguments = '_child', 'table', '_key', 'ascending'
+
+    def _dshape(self):
+        return self._child.dshape
+
+    @property
+    def key(self):
+        return list(self._key)
+
+    def _len(self):
+        return self._child._len()
+
+    @property
+    def _name(self):
+        return self._child._name
+
+    def _copy(self):
+        """Create a copy of self, detached from any specific table."""
+        return Key(None, None, self._key, self.ascending)
+
+    def _invert(self):
+        """Create a copy of self, detached from any specific table and
+           with inverted order."""
+        return Key(None, None, self._key, not self.ascending)
+
+    def __str__(self):
+        func = 'asc' if self.ascending else 'desc'
+        key = repr(self._key[0]) if len(self._key) == 1 else ""
+        return "%s.%s(%s)" % (self._child, func, key)
+
+
+def _isrecord(obj):
+    return isinstance(obj, Expr) and isrecord(obj.dshape.measure)
+
+
+def _find_table_and_field(obj):
+    while True:
+        try:
+            child = obj._child
+        except AttributeError:
+            return None, None
+        if isinstance(obj, Field) and _isrecord(child):
+            return child, obj
+        obj = child
+
+
+def _extract_key_args(first, second, func):
```

Comment:
I don't event think `asc` and `desc` need to be subclasses of `Expr`. They are not expressions. It could even be that `asc(a.b)` returns a `SortKey(field=a.b, ascending=True)` object and `desc(a.b)` returns a `SortKey(field=a.b, ascending=False)` object. I think that making them exprs will only complicate things. How does this sound?